common/src/main/java/com/wynntils/core/net/NetManager.java

```diff
@@ -106,28 +106,40 @@ public void openLink(UrlId urlId, Map<String, String> arguments) {
         openLink(uri);
     }
 
-    private HttpRequest createGetRequest(URI uri) {
-        return HttpRequest.newBuilder()
+    private HttpRequest createGetRequest(URI uri, boolean athenaUrl) {
+        HttpRequest.Builder builder = HttpRequest.newBuilder()
                 .uri(uri)
                 .timeout(Duration.ofMillis(REQUEST_TIMEOUT_MILLIS))
-                .header("User-Agent", USER_AGENT)
-                .build();
+                .header("User-Agent", USER_AGENT);
+
+        // If we are just bootstrapping, we don't have a token (or WynntilsAccountManager) yet
+        if (Managers.WynntilsAccount != null && athenaUrl) {
+            builder.header("authToken", Managers.WynntilsAccount.getToken());
+        }
+
+        return builder.build();
     }
 
-    private HttpRequest createPostRequest(URI uri, JsonObject jsonArgs) {
-        return HttpRequest.newBuilder()
+    private HttpRequest createPostRequest(URI uri, boolean athenaUrl, JsonObject jsonArgs) {
+        HttpRequest.Builder builder = HttpRequest.newBuilder()
                 .uri(uri)
                 .timeout(Duration.ofMillis(REQUEST_TIMEOUT_MILLIS))
                 .header("User-Agent", USER_AGENT)
                 .header("Content-Type", "application/json")
-                .POST(HttpRequest.BodyPublishers.ofString(jsonArgs.toString()))
-                .build();
+                .POST(HttpRequest.BodyPublishers.ofString(jsonArgs.toString()));
+
+        // If we are just bootstrapping, we don't have a token (or WynntilsAccountManager) yet
+        if (Managers.WynntilsAccount != null && athenaUrl) {
```

Comment:
I'm not entirely happy how knowledge about athena auth is hard-coded in the NetManager. I wonder if we should add a `Map<String, String> headers` argument to the NetManager methods instead. At some point of course we need to determine if this is an athena call and if we have a token, but this feels like the wrong place.

Reply:
Even just adding a `createAthenaPostRequest` here that forwards to `createPostRequest` and adds an Athena header would be an improvement. But I think there could be a better place. Let me look around a bit.